### PR TITLE
Add api for ppx_type_conv

### PR DESCRIPTION
**Pull request by rgrinberg**
This consists of two parts:
* A way to fetch all currently registered derivers
* A way to add hooks for when new derivers are registered

@diml is this good enough?

### CODE REVIEW

**Comment by diml** [comment]
Yes, that's enough to use ppx\_deriving plugins with ppx\_type\_conv. Note that `derivers` can probably just return a `deriver list` given that the key is the same as the `name` field

**Comment by rgrinberg** [comment]
@diml sure. I made the fix.

File: src/ppx_deriving.mli [hunk 1]
```diff
@@ -42,6 +42,10 @@ type deriver = {
 (** [register deriver] registers [deriver] according to its [name] field. *)
 val register : deriver -> unit
 
+val add_register_hook : (deriver -> unit) -> unit
```
Comment (whitequark) [inline src/ppx_deriving.mli]: Documentation?

Comment (rgrinberg) [inline src/ppx_deriving.mli]: alright

File: src/ppx_deriving.mli [hunk 2]
```diff
@@ -42,6 +42,13 @@ type deriver = {
 (** [register deriver] registers [deriver] according to its [name] field. *)
 val register : deriver -> unit
 
+(** [add_register_hook hook] adds [hook] to be executed whenever a new deriver
+    is registered *)
+val add_register_hook : (deriver -> unit) -> unit
+
+(** [derivers ()] returns all currently registered derivers *)
```
Comment (whitequark) [inline src/ppx_deriving.mli]: Nitpick: the sentences in the docs are finished with dots. I'll merge after this is fixed!

**Comment by rgrinberg** [comment]
No problem.

**Pull request merged**